app/adapters/threema_adapter/outbound.rb

```diff
@@ -52,13 +52,14 @@ def send_files(files, message)
             file_path: ActiveStorage::Blob.service.path_for(file.attachment.blob.key),
             file_name: file.attachment.blob.filename.to_s,
             caption: index.zero? ? message.text : nil,
-            render_type: :media
+            render_type: :media,
+            message: message
           )
         end
       end
 
       def send_text(message)
-        ThreemaAdapter::Outbound::Text.perform_later(contributor_id: message.recipient.id, text: message.text)
+        ThreemaAdapter::Outbound::Text.perform_later(contributor_id: message.recipient.id, text: message.text, message: message)
```

Comment:
I came to the same conclusion. The gap in understanding on my part was from the fact that these outbound text classes also handle sending text messages that have no message object in the db.

This happens for welcome messages, for example. See https://github.com/tactilenews/100eyes/blob/main/app/adapters/threema_adapter/outbound.rb#L25

Reply:
One final note on this before I merge. With the solution proposed in https://github.com/tactilenews/100eyes/issues/1869, we could easily refactor to send out the welcome message, which going forward should always actually be saved in the db.

This would have the benefits of being able to refactor to always expect a message object be passed in to these outbound classes and we would be able to personalize our welcome message by inserting the contributors first name into the placeholder created in this welcome message request.